benthomasson/eda/tests/unit/event_source/test_webhook.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+import aiohttp
+import asyncio
+
+from plugins.event_source.webhook import main as webhook_main
+
+
+async def start_server(queue, args):
+    await webhook_main(queue, args)
+
+
+async def post_code(server_task, info):
+    async with aiohttp.ClientSession() as session:
+        async with session.post('http://localhost:8000/payload', json=info) as resp:
+            print(resp.status)
+ 
+    server_task.cancel()
+
+
+async def cancel_code(server_task):
+    server_task.cancel()
+
+
+@pytest.mark.asyncio
+async def test_cancel():
+    queue = asyncio.Queue()
+
+    args = {"host": 'localhost', "port": 8000}
+    plugin_task = asyncio.create_task(start_server(queue, args))
+    asyncio.sleep(1)
```

Comment:
@hsong-rh Is this sleep needed? Is it just waiting for the other task to startup

Reply:
The purpose of it is to make sure the `plugin_task` run first

Reply:
I have not looked in the deep the code, but in a first look seems that using sleeps to ensure the order execution in a concurrent context seems a bad idea, I'm afraid to have a flaky test.

Reply:
It's been removed